File: src/ios/PrivateImageViewPlugin.m
#import "PrivateImageViewPlugin.h"

@implementation PrivateImageViewPlugin

- (void)showPrivateImage:(CDVInvokedUrlCommand*)command
{
    NSDictionary* options = [command.arguments objectAtIndex:0];
    UIImage* image = [UIImage imageNamed:[options objectForKey:@"image"]];
    AVLayerVideoGravity contentMode = kCAGravityResizeAspect;
    if ([[options objectForKey:@"contentMode"] isEqualToString:@"resize"]) {
        contentMode = kCAGravityResize;
    } else if ([[options objectForKey:@"contentMode"] isEqualToString:@"center"]) {
        contentMode = kCAGravityCenter;
    }
    BOOL preventCapture = [[options objectForKey:@"preventCapture"] boolValue];
    
    PrivateImageView* imageView = [[PrivateImageView alloc] initWithImage:image contentMode:contentMode preventCapture:preventCapture];
    
    // Add the PrivateImageView to the current view controller's view
    [self.viewController.view addSubview:imageView];
    
    CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:@"Image added"];
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

@end

File: src/ios/PrivateImageViewPlugin.swift
import Foundation
import AVFoundation

@objc(PrivateImageViewPlugin) class PrivateImageViewPlugin : CDVPlugin {
    var privateImageView: PrivateImageView?

    @objc(setImage:)
    func setImage(command: CDVInvokedUrlCommand) {
        guard let imageData = command.arguments[0] as? String else { return }
        guard let data = Data(base64Encoded: imageData) else { return }
        let image = UIImage(data: data)
        privateImageView?.image(image)
    }

    @objc(setImageContentMode:)
    func setImageContentMode(command: CDVInvokedUrlCommand) {
        let mode = command.arguments[0] as? AVLayerVideoGravity
        privateImageView?.imageContentMode(mode)
    }

    @objc(setPreventCapture:)
    func setPreventCapture(command: CDVInvokedUrlCommand) {
        let prevent = command.arguments[0] as? Bool
        privateImageView?.preventCapture(prevent)
    }
}
